packages/starrating/src/react/index.js

```diff
@@ -0,0 +1,155 @@
+import * as glamor from 'glamor'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import { withTheme } from '@pluralsight/ps-design-system-theme/react'
+
+import Star from './star'
+
+import css, { BASE_CLASSNAME } from '../css'
+
+const styles = {
+  starRating: () => glamor.css(css[BASE_CLASSNAME]),
+  screenReaderInput: () =>
+    glamor.css(css[`${BASE_CLASSNAME}__screen-reader-input`]),
+  screenReaderText: () =>
+    glamor.css(css[`${BASE_CLASSNAME}__screen-reader-text`])
+}
+
+const ScreenReaderInput = props => (
+  <input {...styles.screenReaderInput(props)} tabIndex={-1} {...props} />
+)
+
+const ScreenReaderText = props => (
+  <span {...styles.screenReaderText(props)} {...props} />
+)
+
+class StarRating extends React.PureComponent {
+  constructor(props) {
+    super(props)
+
+    this.handleStarClicked = this.handleStarClicked.bind(this)
+    this.handleStarEnter = this.handleStarEnter.bind(this)
+    this.handleStarLeave = this.handleStarLeave.bind(this)
+
+    this.state = { hoverIndex: null, interactive: !!this.props.onChange }
```

Comment:
Does interactive need to be on state?  Doesn't seem to be something that would change much.  I'll keep reading to find out more...

If was not on state, would not need componentWillReceiveProps

Reply:
i could just do the check inline - or use a getter to encapsulate the logic. doing it here is faster...but not in any noticeable way

Reply:
I didn't remember to come back and update this comment, but I'd sure like to not use state if not required. 

I'm thinking of it this way: Prefer props, they're simpler, incoming/immutable data belongs based on react conventions, react state is for internal changing data, which this is not.

It reads well elsewhere (the `interactive` flag is easy to grok).  But why it's on `state` is not easy to understand and seems to be unconventional.  

I'd also be happy to do a refactor after merge.

Reply:
i'll do a quick refactor into a getter and merge. we can sync up after and come up with a good approach